vault/resource_kv_secret_v2.go

```diff
@@ -201,14 +201,15 @@ func kvSecretV2Write(ctx context.Context, d *schema.ResourceData, meta interface
 
 	path := getKVV2Path(mount, name, consts.FieldData)
 
-	var secretData map[string]interface{}
-	err := json.Unmarshal([]byte(d.Get(consts.FieldDataJSON).(string)), &secretData)
-	if err != nil {
-		return diag.Errorf("data_json %#v syntax error: %s", d.Get(consts.FieldDataJSON), err)
-	}
+	data := map[string]interface{}{}
+	if dataJSON, ok := d.GetOk(consts.FieldDataJSON); ok {
```

Comment:
What is the purpose of `ok` here? Since the field is `Required`, wouldn't TF would break early in the validation stage if this was not provided by the user? My understanding is we would never get to the Write function if `ok` was not `true`

Reply:
If this resource is imported then the `FieldDataJSON` is provided on Create via the import. However, on Update this may not be set if the field is set to be ignored via the TF `ignore_changes` lifecycle setting.

Reply:
Never mind, I didn't see the linked issue. I see that this is to enable `ignore_changes` in `lifecycle`